src/util/object.ts

```diff
@@ -0,0 +1,32 @@
+// Utility function to test whether an object has own nested property
+// This is used instead of Object.prototype.hasOwnProperty() due to browser support
+// Accpets object path in dot notation form
+export function hasProperty(obj: any, propertyPath: string): boolean {
```

Comment:
If you keep this util, I would think `hasOwnProperty` with a `try` before it seems more battle-hardened

Reply:
Fixed in https://github.com/fastly/insights.js/pull/28/commits/b3c32dfc575acd74087d48a6b6c969abca0091d2